packages/bierzo-wallet/src/store/reducers.ts

```diff
@@ -15,14 +21,24 @@ export interface RootReducer {
   usernames: UsernamesState;
 }
 
-const createRootReducer = (): Reducer<RootReducer> =>
-  combineReducers({
-    extension: extensionReducer,
-    notifications: notificationReducer,
-    tokens: tokensReducer,
-    balances: balancesReducer,
-    usernames: usernamesReducer,
-  });
+const allReducers = combineReducers({
+  extension: extensionReducer,
+  notifications: notificationReducer,
+  tokens: tokensReducer,
+  balances: balancesReducer,
+  usernames: usernamesReducer,
+});
+
+const createRootReducer = (): Reducer<RootReducer> => (
+  state: RootReducer | undefined,
+  action: ActionType<typeof actions>,
+) => {
+  if (action.type === "@@app/RESET") {
+    state = undefined;
```

Comment:
Reassigning a function argument is not good practice.

But `return allReducers(undefined, action);` would do the job, right?

Reply:
yes it should.